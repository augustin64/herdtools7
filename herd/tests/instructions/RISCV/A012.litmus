RISCV A012
{
0:x6=x; 0:x9=y; 0:x12=z;
1:x6=z; 1:x8=x;
}
 P0            | P1             ;
 lw x5,0(x6)   | lw x5,0(x6)    ;
 xor x7,x5,x5  | ori x7,x0,1    ;
 ori x8,x0,1   | sw.rl x7,0(x8) ;
 add x10,x9,x7 |                ;
 sw x8,0(x10)  |                ;
 ori x11,x0,1  |                ;
 sw x11,0(x12) |                ;
~exists (0:x5=1 /\ 1:x5=1)
