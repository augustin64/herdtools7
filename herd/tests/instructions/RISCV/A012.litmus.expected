Test A012 Forbidden
States 3
0:x5=0; 1:x5=0;
0:x5=0; 1:x5=1;
0:x5=1; 1:x5=0;
Ok
Witnesses
Positive: 3 Negative: 0
Condition ~exists (0:x5=1 /\ 1:x5=1)
Observation A012 Never 0 3
Hash=df47ef7bc9c12d0239f38fcd2a9a6d3f

